client/app/bundles/course/reference-timelines/components/SubmitIndicator.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from 'react';
+import { Cancel, CheckCircle } from '@mui/icons-material';
+import { Chip, Grow, Tooltip, Typography } from '@mui/material';
+import moment from 'moment';
+
+import LoadingIndicator from 'lib/components/core/LoadingIndicator';
+import useTranslation from 'lib/hooks/useTranslation';
+
+import { useLastSaved } from '../contexts';
+import translations from '../translations';
+
+const RELATIVE_TIME_UPDATE_INTERVAL_MS = 60000 as const;
+const ANNOUNCE_ANIMATION_DURATION_MS = 500 as const;
+const ANNOUNCE_FLASH_DURATION_MS = 2000 as const;
+
+interface SubmitIndicatorProps {
+  className?: string;
+}
+
+interface SavedIndicatorProps {
+  at: moment.Moment;
+  success: boolean;
+}
+
+const SavedIndicator = (props: SavedIndicatorProps): JSX.Element => {
+  const { at: lastSaved, success } = props;
+
+  const { t } = useTranslation();
+
+  const [announcing, setAnnouncing] = useState(true);
+  const [relativeTime, setRelativeTime] = useState('');
+
+  const updateRelativeDescription = (): void =>
+    setRelativeTime(lastSaved.fromNow());
+
+  useEffect(() => {
+    if (announcing) {
+      const timeout = setTimeout(() => {
+        setAnnouncing(false);
+        updateRelativeDescription();
+      }, ANNOUNCE_FLASH_DURATION_MS);
+
+      return () => clearTimeout(timeout);
+    }
+
+    const timer = setInterval(
+      updateRelativeDescription,
+      RELATIVE_TIME_UPDATE_INTERVAL_MS,
+    );
+
+    return () => clearInterval(timer);
+  }, [announcing]);
+
+  return (
+    <>
+      {announcing && (
```

Comment:
It still doesn't work. Note that the animation I was mentioning was not the animation of "Saved" coming up, but the outgoing of "Saved" and incoming of "Last saved at ...". Otherwise, the latter will not grow-out/grow-in, but just "pop" cuts and changes.

Reply:
<img width="599" alt="image" src="https://user-images.githubusercontent.com/42570513/212821808-7d3e0d06-5312-421b-b25c-c810ed0735e7.png">

Reply:
I see, so you meant having different keys. I thought you meant the same for both. Yup, this works, too. I have since added the keys, but without the fragment in your screenshot.